From 20d56cfc56000194d2bcf90f57b3319eaa472af5 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 7 Apr 2018 06:33:43 +0200
Subject: [PATCH 1/2] build: move usp10 before gdi32

Without this pango on mingw64 tries to lookup up the Script* functions
in gdi32 and fails. It already fails at the build stage because the
introspection dump crashes with a missing entry point error.

Moving usp10 before gdi32 makes things work.

This might be related to the warning in the uniscribe docs:
"Important Starting with Windows 8: To maintain the ability to run on
Windows 7, a module that uses Uniscribe must specify Usp10.lib
before gdi32.lib in its library list."
---
 pango/Makefile.am | 2 +-
 pango/meson.build | 2 +-
 pangowin32.pc.in  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/pango/Makefile.am b/pango/Makefile.am
index 0ce84357..b360533d 100644
--- a/pango/Makefile.am
+++ b/pango/Makefile.am
@@ -393,7 +393,7 @@ libpangowin32_1_0_la_LDFLAGS = $(LIBRARY_LIBTOOL_OPTIONS)
 libpangowin32_1_0_la_LIBADD =			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(GLIB_LIBS)				\
-	-lgdi32 -lusp10
+	-lusp10 -lgdi32
 libpangowin32_1_0_la_DEPENDENCIES =		\
 	libpango-$(PANGO_API_VERSION).la
 libpangowin32_1_0_la_SOURCES =	\
diff --git a/pango/meson.build b/pango/meson.build
index 3312d72d..42fd7e0d 100644
--- a/pango/meson.build
+++ b/pango/meson.build
@@ -306,8 +306,8 @@ if host_system == 'windows'
 
   pangowin32_deps = pango_deps + [
     libpango_dep,
-    cc.find_library('gdi32'),
     cc.find_library('usp10'),
+    cc.find_library('gdi32'),
   ]
 
   pangowin32_rc = configure_file(
diff --git a/pangowin32.pc.in b/pangowin32.pc.in
index 4c9b879f..34ea6a3c 100644
--- a/pangowin32.pc.in
+++ b/pangowin32.pc.in
@@ -8,5 +8,5 @@ Description: Win32 GDI font support for Pango
 Version: @VERSION@
 Requires: pango
 Libs: -L${libdir} -lpangowin32-@PANGO_API_VERSION@
-Libs.private: -lgdi32 -lusp10
+Libs.private: -lusp10 -lgdi32
 Cflags: -I${includedir}/pango-1.0
-- 
2.16.2

